tests/integration/acp/relationship/add_doc_actor_test/add_doc_actor_with_manager_gql_test.go

```diff
@@ -0,0 +1,604 @@
+
+			testUtils.AddDocActorRelationship{ // Manager can manger only.
+				RequestorIdentity: 2,
+
+				TargetIdentity: 3,
+
+				CollectionID: 0,
+
+				DocID: 0,
+
+				Relation: "reader",
+
+				ExpectedExistence: false,
+			},
+		},
+	}
+
+	testUtils.ExecuteTestCase(t, test)
+}
+
+func TestACP_ManagerCanNotGivesAccessNotDefinedInPolicyToAnotherActor_GQL_Error(t *testing.T) {
+	expectedPolicyID := "fc56b7509c20ac8ce682b3b9b4fdaad868a9c70dda6ec16720298be64f16e9a4"
+
+	test := testUtils.TestCase{
+
+		Description: "Test acp, owner tries to give access to another actor not defined in policy",
+
+		SupportedMutationTypes: immutable.Some([]testUtils.MutationType{
+			// GQL mutation will return no error when wrong identity is used so test that separately.
+			testUtils.GQLRequestMutationType,
+		}),
+
+		Actions: []any{
+			testUtils.AddPolicy{
+
+				Identity: immutable.Some(1),
+
+				Policy: `
+                    name: Test Policy
+
+                    description: A Policy
+
+                    actor:
+                      name: actor
+
+                    resources:
+                      users:
+                        permissions:
+                          read:
+                            expr: owner + reader + writer
+
+                          write:
+                            expr: owner + writer
+
+                          nothing:
+                            expr: dummy
+
+                        relations:
+                          owner:
+                            types:
+                              - actor
+
+                          reader:
+                            types:
+                              - actor
+
+                          writer:
+                            types:
+                              - actor
+
+                          admin:
+                            manages:
+                              - reader
+                            types:
+                              - actor
+
+                          dummy:
+                            types:
+                              - actor
+                `,
+
+				ExpectedPolicyID: expectedPolicyID,
+			},
+
+			testUtils.SchemaUpdate{
+				Schema: fmt.Sprintf(`
+						type Users @policy(
+							id: "%s",
+							resource: "users"
+						) {
+							name: String
+							age: Int
+						}
+					`,
+					expectedPolicyID,
+				),
+			},
+
+			testUtils.CreateDoc{
+				Identity: immutable.Some(1),
+
+				CollectionID: 0,
+
+				Doc: `
+					{
+						"name": "Shahzad",
+						"age": 28
+					}
+				`,
+			},
+
+			testUtils.AddDocActorRelationship{ // Make admin / manager
+				RequestorIdentity: 1,
+
+				TargetIdentity: 2,
+
+				CollectionID: 0,
+
+				DocID: 0,
+
+				Relation: "admin",
+
+				ExpectedExistence: false,
+			},
+
+			testUtils.AddDocActorRelationship{ // Admin tries to make another actor a writer
+				RequestorIdentity: 2,
+
+				TargetIdentity: 3,
+
+				CollectionID: 0,
+
+				DocID: 0,
+
+				Relation: "writer",
```

Comment:
question: I don't see how this triggers the error. The title of the test says "CanNotGivesAccessNotDefinedInPolicy", but "writer" IS defined in the policy.

Reply:
That is because a manager can not give access to another actor with a relation that it does not manage.

Good shout on the test name, will change to something better.

- [ ] Better test name